### PR TITLE
individual market osse plan shopping and migrations

### PR DESCRIPTION
**Pull request by raghuramg**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [ ] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: 

# A brief description of the changes

Current behavior:

New behavior:

Updated migration code...
- Fixed eligibility persistence issues with IVL benefit coverage period.
- renew eligibility as well, when they have active eligibility at the end of 2022. 
- Also add code changes to handle eligibility record that span across multiple years.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [x] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

### CODE REVIEW

File: app/domain/operations/eligible/migrate_eligibility.rb [hunk 1]
```diff
@@ -40,10 +40,48 @@ def find_subject(values)
         Success(subject)
       end
 
+      def add_renewal_eligibilities(values)
+        prev_year = Date.today.year - 1
+        eligibility_years =
+          values[:current_eligibilities].collect(&:start_on).map(&:year).uniq
+        return unless eligibility_years == [prev_year]
```
Comment (nks2109) [inline app/domain/operations/eligible/migrate_eligibility.rb:47]: Here do you get multiple years or any single year?
If yes then you might need to check this way right?
```
eligibility_years.include?(prev_year)
```

Comment (nks2109) [reply app/domain/operations/eligible/migrate_eligibility.rb:47]: if you get multiple years then are you checking whether a single previous year is present in the array or do you want to see whether that previous year is included in that array? @raghuramg

**Review by nks2109 — APPROVED** [review]

File: lib/tasks/migrations/plans/load_dc_benefit_market_catalog.rake [hunk 2]
```diff
@@ -40,6 +40,26 @@ namespace :load do
...
+          {
+            subject: benefit_market_catalog.to_global_id,
+            eligibility_feature: "aca_shop_osse_eligibility",
+            effective_date: benefit_market_catalog.application_period.min.to_date,
+            domain_model: "AcaEntities::BenefitSponsors::BenefitSponsorships::BenefitSponsorship"
+          }
+        )
+
+        if result.success?
+          p "Success: created eligibility for #{current_period_bc.start_on.year} benefit coverage period"
```
Comment (nks2109) [inline lib/tasks/migrations/plans/load_dc_benefit_market_catalog.rake:56]: here we don't get **current_period_bc** value. We do get benefit_market_catalog

File: lib/tasks/migrations/create_ivl_benefit_packages_DC.rake [hunk 3]
```diff
@@ -322,5 +322,21 @@ namespace :import do
...
 
     current_period_bc.save!
+
+    puts "current benefit coverage period created. Creating eligibilities....."
+    result = Operations::Eligible::CreateCatalogEligibility.new.call(
+      {
+        subject: current_period_bc.to_global_id,
+        eligibility_feature: "aca_ivl_osse_eligibility",
+        effective_date: current_period_bc.start_on.to_date,
+        domain_model: "AcaEntities::BenefitSponsors::BenefitSponsorships::BenefitSponsorship"
```
Comment (nisanthyaganti9) [inline lib/tasks/migrations/create_ivl_benefit_packages_DC.rake:332]: do we use shop class as domain model for ivl catalog creation ?

File: lib/tasks/migrations/plans/load_dc_benefit_market_catalog.rake (hunk 2 above)
Comment (nisanthyaganti9) [reply lib/tasks/migrations/plans/load_dc_benefit_market_catalog.rake:56]: +1

**Pull request closed without merging**